scripts/optimize_bayesopt.py

```diff
@@ -0,0 +1,154 @@
+import subprocess
+import os
+import json
+import docker
+import shutil
+from pathlib import Path
+from bayes_opt import BayesianOptimization
+from bayes_opt.logger import JSONLogger
+from bayes_opt.event import Events
+from bayes_opt.util import load_logs
+
+
+default_parameters = {
```

Comment:
can we avoid this and read it from file/ or import from the HTM detector?

Reply:
mh not really, as the params are "locked into" the docker file. I think importing it would reduce readability and understandability. I see it like this: A user who wants to run param optimization on NAB just created the docker image from the provided docker file and can easily run his/her own optimization by looking at the examples provided. Note that the scripts should be just examples on how to run param opt on NAB.

Reply:
>  A user who wants to run param optimization on NAB just created the docker image from the provided docker file and can easily run his/her own optimization 

so a user builds a docker image and runs optimization from that? I thought it's you run the script/opt and it spawns docker-instances for each detector-params combination ?

Reply:
> so a user builds a docker image and runs optimization from that?

No, he builds the image from the Dockerfile we provide and the opt script spawns one container for each set of param that is suggested by the opt algorithm, so you're right. I still think that its more convenient if we put the params into the script as (1) the htm.core opt framework requires it and (2) its more clear for the user to work with.

Reply:
Maybe we can get a third opinion on this?

Reply:
Definitely. Or we can leave it as is, I just want to get it working, no strong opinion on this.